azkaban-common/src/main/java/azkaban/user/UserUtils.java

```diff
@@ -22,4 +33,56 @@ public static boolean hasPermissionforAction(final UserManager userManager, fina
 
     return false;
   }
+
+  /**
+   * Creates a watch thread which listens to specified files' modification and reloads
+   * configurations
+   */
+  public static void setupWatch(final String fileName, Logger log, ParseConfigFile parser) {
+    Runnable runnable = () -> {
+      WatchService watchService;
```

Comment:
should we close the WatchService in web server's shutdown hook?

Reply:
We may, although when the server goes down, so will the WatchService.

Reply:
How did you test this behavior?
Will the watch thread prevent the process from existing?

Reply:
I ran solo server manually and it came down without any issue.

Reply:
I am curious about how it worked. This is not a daemon thread, is it?